forc-pkg/src/manifest/manager.rs

```diff
@@ -0,0 +1,484 @@
+    let dep_spec: DepSpec = raw.parse()?;
+    let dep_name = dep_spec
+        .name
+        .clone()
+        .ok_or_else(|| anyhow!("Missing name in `{}`", raw))?;
+
+    let mut details = DependencyDetails {
+        version: dep_spec.version_req.clone(),
+        namespace: None,
+        path: opts.source_path.clone(),
+        git: opts.git.clone(),
+        branch: opts.branch.clone(),
+        tag: opts.tag.clone(),
+        package: None,
+        rev: opts.rev.clone(),
+        ipfs: opts.ipfs.clone(),
+    };
+
+    details.validate()?;
+
+    let dependency_data = if let Some(version) = dep_spec.version_req {
+        Dependency::Simple(version)
+    } else {
+        if details.is_source_empty() {
+            if let Some(member) = member_manifests.get(&dep_name) {
+                let rel_path = pathdiff::diff_paths(member.dir(), package_dir)
+                    .unwrap_or_else(|| member.path().to_path_buf());
+                details.path = Some(rel_path.to_string_lossy().to_string());
+            }
+        }
+        Dependency::Detailed(details)
+    };
+
+    Ok((dep_name, dependency_data))
+}
+/// Reference to a package to be added as a dependency.
+///
+/// See `forc add` help for more info.
+#[derive(Clone, Debug, Default)]
+pub struct DepSpec {
+    pub name: Option<String>,
+    pub version_req: Option<String>,
+}
+
+impl FromStr for DepSpec {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        if s.is_empty() {
+            bail!("dependency spec cannot be empty");
+        }
+
+        let mut dep = DepSpec::default();
+
+        let mut s = s.split('@');
+        let Some(name) = s.next() else {
+            bail!("dependency name is missing");
+        };
+
+        dep.name = Some(name.parse()?);
+
+        let Some(version_req) = s.next() else {
+            return Ok(dep);
+        };
+
+        dep.version_req = Some(version_req.parse()?);
+
+        Ok(dep)
+    }
+}
+
+impl fmt::Display for DepSpec {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Some(name) = &self.name {
+            write!(f, "{name}")?;
+        }
+
+        if let Some(version_req) = &self.version_req {
+            write!(f, "@{version_req}")?;
+        }
+
+        Ok(())
+    }
+}
+
+pub enum DepSection<'a> {
+    Regular(&'a mut BTreeMap<String, Dependency>),
+    Contract(&'a mut BTreeMap<String, ContractDependency>, Option<String>),
+}
+
+impl DepSection<'_> {
+    pub fn insert_dep(&mut self, name: String, data: Dependency, salt: Option<String>) {
+        match self {
+            DepSection::Regular(map) => {
+                map.insert(name, data);
+            }
+            DepSection::Contract(map, salt_opt) => {
+                let resolved_salt = salt
+                    .or_else(|| salt_opt.clone())
+                    .map(|s| HexSalt::from_str(&s).unwrap())
+                    .unwrap_or_else(|| HexSalt(fuel_tx::Salt::default()));
```

Comment:
unnecessary clone; also need to handle `unwrap`
```suggestion
                let resolved_salt = match salt.as_ref().or_else(|| salt_opt.as_ref()) {
                    Some(s) => HexSalt::from_str(s)
                        .map_err(|e| anyhow!("Invalid salt format: {}", e))?,
                    None => HexSalt(fuel_tx::Salt::default()),
                };
```

Reply:
Okay updated function to return result.
Resolved